src/components/LiquidityMining/Create/Steps/RewardAmount/index.tsx

```diff
@@ -101,16 +92,16 @@ export default function RewardsSelection({ rewardsObject, setRewardsObject }: Re
         },
       })
     },
-    [setRewardsObject, rewardsObject]
+    [setRewardsArray, rewardsArray]
   )
 
   return (
     <>
-      <FlexWrapper marginTop="32px" flexWrap="wrap">
-        {[...Array(numberOfRewards)].map((item, index) => (
+      <FlexWrapper justifyContent="center" marginTop="32px" flexWrap="wrap">
+        {[...Array(rewardsArray.length)].map((item, index) => (
```

Comment:
Why not use `rewardsArray.map(item)` directly and then instead `rewardsArray[index]?.reward?.token` we can do `item?.reward?.token`

 Seem odd to use `map` and then `rewardsArray[index]`

Reply:
yeah 100% It made sense with previous setup this is much cleaner!